libvast/test/system/partition.cpp

```diff
@@ -87,4 +91,54 @@ TEST(load) {
       });
 }
 
+TEST(active_partition - create) {
+  // Setup.
+  using namespace std::string_literals;
+  auto id = vast::uuid::random();
+  auto accountant = vast::system::accountant_actor{};
+  auto filesystem = self->spawn(mock_filesystem);
+  auto index_opts = caf::settings{};
+  auto synopsis_opts = index_config{
+    .rules = {
+      {
+        .targets = {"suricata.http.http.url"s},
+        .fp_rate = 0.001,
+      },
+      {
+        .targets = {":address", ":string"},
+        .fp_rate = 0.01,
+      },
+    },
+  };
+  auto store_actor = vast::system::store_actor{};
+  auto store_name = "dummy_store";
+  auto store_header = nullptr;
+  auto active_partition
+    = self->spawn(::vast::system::active_partition, id, accountant, filesystem,
+                  index_opts, synopsis_opts, store_actor, store_name,
+                  store_header);
```

Comment:
Is there a reason you chose to test this with a full partition instead of just the partition_synopsis in isolation?

Reply:
No particular reason, I started with the index to get an almost-end to end test, but the API was a bit too involved. I just use the partition synopsis now.